api/proxy/common/eigenda_network.go

```diff
@@ -104,3 +112,43 @@ func EigenDANetworkFromString(inputString string) (EigenDANetwork, error) {
 			inputString, strings.Join(allowedNetworks, ", "))
 	}
 }
+
+// BuildEthClient creates an Ethereum client using the provided RPC URL and, if set, validates that the chain ID
+// matches the expected EigenDA network. It returns an ethClient, it's ChainID, and an error.
+func BuildEthClient(ctx context.Context, log logging.Logger, ethRpcUrl string,
```

Comment:
this feels like the wrong file to host this function.
Guessing you couldnt keep it in the storage_manager_builder because of circular dependency issues?
Does the monorepo not have another ethClient builder we could reuse? And then we can add our few checks and query for chainID on top?

No need to make a huge refactor but at least I feel like this should be moved to some other file, perhaps under api/proxy/common/eth_client.go or whatever until we find it a better home.